MHD documentation update

armulrich opened the pull request:
Made changes to docs. Added documentation for quasi-symmetry objective and clarified extra_data type in surface.to_vtk
<img width="807" alt="Screenshot 2025-05-11 at 15 19 29" src="https://github.com/user-attachments/assets/c5ee8219-2b33-4c7c-a617-1754350e02ff" />

akaptano reviewed (commented):
Looks good but some tiny notes to add:
1. Please remove the "enumerate" (and js) in the for loop, which goes unused right now.
2. Make it more clear in the normalization description that N(s) is the normalization factor, and in the first case is equal to B_00.
3. Please add the expected output shape of "symmetry_error" depending on the weight choice.

mishapadidar commented on src/simsopt/mhd/boozer.py:
```diff
@@ -295,12 +295,34 @@ def recompute_bell(self, parent=None):
+
+        1. Extracts the Boozer-coordinate Fourier spectrum ``B_{m,n}(s)``.
+        2. Identifies which ``(m,n)`` modes break the specified quasi-symmetry.
+        3. Normalizes all mode amplitudes by either:
```
Computes the normalized amplitudes :math:`A_{m,n}(s)=B_{m,n}(s)/N(s)` where the normalization is either:

mishapadidar commented on src/simsopt/mhd/boozer.py:
```diff
@@ -295,12 +295,34 @@ def recompute_bell(self, parent=None):
+        2. Identifies which ``(m,n)`` modes break the specified quasi-symmetry.
+        3. Normalizes all mode amplitudes by either:
+
+            - :math:`A_{m,n}(s)=B_{m,n}(s)/N(s)` The ``(0,0)`` harmonic (mean field) if ``normalization="B00"``; or
```
- The ``(0,0)`` harmonic (mean field), :math:`N(s) = B_{0,0}`, if ``normalization="B00"``; or

mishapadidar commented on src/simsopt/mhd/boozer.py:
```diff
@@ -295,12 +295,34 @@ def recompute_bell(self, parent=None):
+        4. Applies one of the following three weighting options to the non-symmetric modes, where the error terms for the flux surface s are defined as:
+
+            - ``"even"`` :math:`j_{m,n}=A_{m,n}(s)` returns each normalized amplitude unchanged
+            - ``"stellopt"`` :math:`j_{m,n}=A_{m,n}(s)/s^2` divides each amplitude by ``s_used**2`` (i.e. flux surface number squared) to amplify core-surface errors
```
remove the bit about `s_used` since it is not relevant for users, i.e.,
```
``"stellopt"`` :math:`j_{m,n}=A_{m,n}(s)/s^2` divides each amplitude by flux surface label squared to amplify core-surface errors
```

mishapadidar commented on src/simsopt/geo/surface.py:
```diff
@@ -293,7 +293,7 @@ def to_vtk(self, filename, extra_data=None):
         Args:
             filename: Name of the file to write
-            extra_data: An optional data field on the surface, which can be associated with a colormap in Paraview.
+            extra_data: An optional data field (dictionary) on the surface, which can be associated with a colormap in Paraview.
```
Change to,
```
filename (str): Name of the file to write
extra_data (dict): An optional data field (dictionary) on the surface, which can be associated with a colormap in Paraview.
```

armulrich commented:
Thank you both for the feedback! I’ve addressed all of @akaptano’s and @mishapadidar’s comments, and added a note clarifying the MPI behavior of the function that I previously missed. Please let me know if there’s anything else you’d like me to adjust or that I missed.
<img width="738" alt="Screenshot 2025-05-18 at 00 48 12" src="https://github.com/user-attachments/assets/9bb5bd5e-6af0-48a3-8088-76ea732913f3" />

mishapadidar reviewed (approved):
looks good, thanks.

armulrich commented:
> looks good, thanks.

Great! Happy to help.
